File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Az_Attivita/Models/az-attivita-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";

export class Az_AttivitaModel {
  id: number = 0;
  idAz_Anagrafica: number = 0;
  descrizione: string = '';
}

export class Az_Attivita_GetAll_InModel {}

export class Az_Attivita_GetAll_OutModel extends ModelResult {
  public az_Attivita: Az_AttivitaModel[] = [];
}

export class Az_AttivitaGetInModel { id: number = 0; }
export class Az_AttivitaGetOutModel { az_Attivita: Az_AttivitaModel = new Az_AttivitaModel(); }
export class Az_AttivitaPutInModel { az_Attivita: Az_AttivitaModel = new Az_AttivitaModel(); }
export class Az_AttivitaPutOutModel { az_Attivita: Az_AttivitaModel = new Az_AttivitaModel(); }

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Az_AttivitaService/Models/Az_AttivitaModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_AttivitaService.Models
{
    public class Az_AttivitaModel
    {
        public int Id { get; set; }
        public int IdAz_Anagrafica { get; set; }
        public string Descrizione { get; set; } = string.Empty;
               
        public string? BackgroundColor { get; set; }

        public string? TextColor { get; set; }
    }

    public class Az_Attivita_GetAll_InModel { }
    public class Az_Attivita_GetAll_OutModel : ModelResult
    {
        public List<Az_AttivitaModel> Az_Attivita { get; set; } = new List<Az_AttivitaModel>();
        public Az_Attivita_GetAll_OutModel() { }
    }
    public class Az_AttivitaGetInModel { public int Id { get; set; } = 0; }
    public class Az_AttivitaGetOutModel : ModelResult
    {
        public Az_AttivitaModel Az_Attivita { get; set; } = new Az_AttivitaModel();
    }
    public class Az_AttivitaPutInModel : ModelResult
    {
        public Az_AttivitaModel Az_Attivita { get; set; } = new Az_AttivitaModel();
    }
    public class Az_AttivitaPutOutModel : ModelResult
    {
        public Az_AttivitaModel Az_Attivita { get; set; } = new Az_AttivitaModel();
    }
}
